geniza/corpus/views.py

```diff
@@ -655,25 +728,21 @@ class DocumentTranscribeView(PermissionRequiredMixin, DocumentDetailView):
     viewname = "corpus:document-transcribe"
 
     def page_title(self):
-        # Translators: title of transcription editor page
-        return _("Edit transcription for %(doc)s") % {"doc": self.get_object().title}
+        """Title of transcription editor page"""
+        return "Edit transcription for %(doc)s" % {"doc": self.get_object().title}
 
     def get_context_data(self, **kwargs):
         """Pass annotation configuration and TinyMCE API key to page context"""
         context_data = super().get_context_data(**kwargs)
 
-        # get source uri if source_pk present in kwargs (i.e. editing an existing transcription),
-        # and the source exists
         source = None
-        source_pk = self.kwargs.get("source_pk", None)
-        # if source_pk is None, it's a new transcription, so no Source query
-        # TODO: once add transcription view is separated, this view should not be called without
-        # source_pk, so we should redirect there if source_pk is none
-        if source_pk is not None:
-            try:
-                source = Source.objects.get(pk=source_pk)
-            except Source.DoesNotExist:
-                raise Http404
+        # source_pk will always be an integer here; otherwise, a different (or no) route
+        # would have matched
+        source_pk = self.kwargs.get("source_pk")
+        try:
+            source = Source.objects.get(pk=source_pk)
```

Comment:
could consolidate if we aren't using `source_pk` anywhere else

```suggestion
        try:
            source = Source.objects.get(pk=self.kwargs["source_pk"])
```